test/XRP/fakes/fake-xrp-protobufs.ts

```diff
@@ -106,36 +173,106 @@ path2.addElements(path2Element1)
 const pathList = [path1, path2]
 
 // Payments
-
 const paymentMandatoryFields = new Payment()
 paymentMandatoryFields.setDestination(destination)
-paymentMandatoryFields.setAmount(amount)
+paymentMandatoryFields.setAmount(amountXrp)
+
+const paymentMandatoryFieldsIssuedCurrency = new Payment()
+paymentMandatoryFieldsIssuedCurrency.setDestination(destination)
+paymentMandatoryFieldsIssuedCurrency.setAmount(amountIssuedCurrency)
 
 const paymentAllFields = new Payment()
 paymentAllFields.setDestination(destination)
-paymentAllFields.setAmount(amount)
+paymentAllFields.setAmount(amountXrp)
 paymentAllFields.setDestinationTag(destinationTag)
 paymentAllFields.setInvoiceId(invoiceId)
 paymentAllFields.setDeliverMin(deliverMin)
 paymentAllFields.setSendMax(sendMax)
 paymentAllFields.setPathsList(pathList)
 
-// Transactions
+// Transaction
+
+/**
+ * Helper function to generate Transaction objects with the standard values from Payment objects.
+ *
+ * @param payment -Payment object to insert into the transaction.
+ * @returns Payment Transaction with the included payment param.
+ */
+function buildStandardTransactionFromPayment(payment: Payment): Transaction {
```

Comment:
this is nice.  might want to refactor it (perhaps in later PR) to build a transaction of generic type and be able to feed it whichever transaction type (AccountSet, TrustSet etc.)

Reply:
Or something like `populateCommonTransactionFields` and then add the specific transaction data to the Transaction object after it's been returned